### PR TITLE
fix: return clean error messages instead of raw SDK objects

### PR DESCRIPTION
**Pull request by dylankilkenny**
## Summary
- Check `simResult.error` before calling `assembleTransaction()` to prevent massive JSON dumps with `_attributes`/`_switch`/`_value` being exposed to clients
- Add `parseSimulationError()` to extract human-readable messages from Soroban diagnostic events (e.g., "signature has expired (Auth, InvalidInput)")
- Add `sanitizeReason()` to strip provider wrapper text from transaction errors (e.g., "TxInsufficientBalance" instead of nested error chain)
- Add comprehensive tests for both error sanitization functions

## Test plan
- [x] Unit tests added for `parseSimulationError()` and `sanitizeReason()`
- [ ] Manual testing against local relayer with expired signatures
- [ ] Verify error responses are clean JSON without internal SDK objects

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error messaging for simulation failures with more human-readable and diagnostic information
  * Enhanced error messages for on-chain transaction failures with clearer, sanitized error details
  * Added better logging for RPC errors to support debugging and troubleshooting

* **Tests**
  * Added comprehensive unit tests covering error handling scenarios, edge cases, and message sanitization

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/plugin/simulation.ts [hunk 1]
```diff
@@ -65,33 +65,36 @@ export async function simulateAndBuildWithChannel(
 
   if (rpcResponse.error) {
     const { code, message, description, data } = rpcResponse.error as any;
-    throw pluginError('Simulation RPC execution failed', {
+    console.error(`[channels] RPC error: code=${code}, message=${message}, detail=${description || data}`);
+    throw pluginError('Simulation RPC failed', {
       code: 'SIMULATION_RPC_FAILURE',
+      status: HTTP_STATUS.BAD_GATEWAY,
+      details: { message: 'RPC provider error' },
```
[the remark this answers is not in the document]
Comment (tirumerla) [reply src/plugin/simulation.ts]: @copilot details for the rpc error is added to console.error above and want to keep this generic as to not expose details to the client

File: src/plugin/simulation.ts [hunk 2]
```diff
@@ -65,33 +65,36 @@ export async function simulateAndBuildWithChannel(
...
+  if ('error' in simResult && simResult.error) {
+    throw pluginError('Simulation failed', {
+      code: 'SIMULATION_FAILED',
       status: HTTP_STATUS.BAD_REQUEST,
-      details: {
-        rpcCode: code,
-        message,
-        description: description || data,
-      },
+      details: { error: parseSimulationError(simResult.error) },
```
Comment (tirumerla) [inline src/plugin/simulation.ts]: Should we add console.error log here

File: src/plugin/simulation.ts [hunk 3]
```diff
@@ -65,33 +65,36 @@ export async function simulateAndBuildWithChannel(
...
     // Use SDK's assembleTransaction to apply simulation results
     const prepared = rpc.assembleTransaction(transaction, simResult).build() as Transaction;
 
     const resourceFee = prepared.toEnvelope().v1().tx().ext().sorobanData()?.resourceFee();
     console.debug(`[channels] Simulation complete: resourceFee=${resourceFee}`);
     return prepared;
   } catch (err: any) {
-    throw pluginError('Simulation result processing failed', {
+    throw pluginError('Simulation failed', {
       code: 'SIMULATION_FAILED',
```
Comment (tirumerla) [inline src/plugin/simulation.ts]: Same here
